python/dolfinx/fem/__init__.py

```
@@ -34,6 +46,30 @@ def create_sparsity_pattern(a: Form):
     return _create_sparsity_pattern(a._cpp_object)
 
 
+def create_nonmatching_meshes_interpolation_data(
+    mesh_to: typing.Union[_Mesh, typing.Union[_Geometry_float64, _Geometry_float32]],
+    element: typing.Union[_FiniteElement_float32, _FiniteElement_float64],
+    mesh_from: _Mesh, cells: typing.Optional[npt.NDArray[np.int32]] = None, padding: float = 1e-14
+):
+    """Generate data needed to interpolate discrete functions across different meshes.
+
+    Args:
+        mesh_to: Mesh or geometry to interpolate into
+        element: Element of the space to interpolate into
+        mesh_from: Mesh of the function to interpolate from
+        cells: Indices of the cells in the destination mesh on which to interpolate.
+        padding: Absolute padding of bounding boxes of all entities on mesh_to
+    """
+    if cells is None:
+        assert isinstance(mesh_to, _Mesh)
```

I don't think the assert is necessary - just let an exception be raised.